x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java

```diff
@@ -273,6 +273,9 @@ protected final void doTest(String query) throws Throwable {
         Map<?, ?> prevTooks = supportsTook() ? tooks() : null;
         Map<String, Object> answer = runEsql(builder.query(query), testCase.assertWarnings(deduplicateExactWarnings()));
 
+        var isPartial = answer.get("is_partial");
+        assertTrue("unexpected partial results", isPartial == null || (boolean) isPartial == false);
```

Comment:
Might be easier to read with `assertThat(..., anyOf(is(false), nullValue()))`. Dunno. Your call.